### PR TITLE
Use Linux-built cache/sysroot for Mac CI tester 

### PR DESCRIPTION
**Pull request by dschuff**
This PR makes the CircleCI builders use the sysroot built by the Linux builder on the mac bot, instead of having the mac bot rebuild all the libraries. It does this by attaching the same workspace that the linux testers do, and then removing the Linux-specific binaries and emsdk, and using the mac EMSDK instead.

### CODE REVIEW

**Comment by dschuff** [comment]
I'm curious to hear what you think of this. It saves about 37% of the mac bot time without reducing the test coverage (other than any coverage that comes from building the libraries themselves).

File: .circleci/config.yml [hunk 1]
```diff
@@ -568,7 +569,13 @@ jobs:
             brew list ninja || brew install ninja
             brew list python3 || brew install python3
             brew list pkg-config || brew install pkg-config
+      - attach_workspace:
+          at: ~/
       - checkout
+      - run:
+          name: Remove native binaries
```
Comment (sbc100) [inline .circleci/config.yml:576]: Remove linux binaries?

What is vms?

Comment (dschuff) [reply .circleci/config.yml:576]: `vms` is where native VM runtimes such as v8 and wasmtime are downloaded (see e.g. [here](https://app.circleci.com/pipelines/github/emscripten-core/emscripten/24115/workflows/97c4fd94-24fe-4cec-a142-81a38ebb3146/jobs/575269) where it tried to run wasmtime left over from the Linux bot.

File: .circleci/config.yml [hunk 2]
```diff
@@ -94,19 +94,19 @@ commands:
...
             test -f ~/vms/wasmer && echo "WASMER = os.path.expanduser(os.path.join('~', 'vms', 'wasmer')) ; WASM_ENGINES.append(WASMER)" >> .emscripten || true
-            test -d ~/wasi-sdk && cp -ar ~/wasi-sdk/lib/ ~/emsdk/upstream/lib/clang/16.0.0/
+            test -d ~/wasi-sdk && cp -a ~/wasi-sdk/lib/ ~/emsdk/upstream/lib/clang/16.0.0/
             cd -
             echo "final .emscripten:"
             cat ~/emsdk/.emscripten
       - emsdk-env
       - npm-install
+  build-libs:
+    description: "Build all libraries, and freeze the cache"
```
Comment (kripken) [inline .circleci/config.yml:104]:
```suggestion
    description: "Build all libraries"
```
Looks like this comment was stale/incorrect before, but might as well fix it now.

File: .circleci/config.yml [hunk 3]
```diff
@@ -628,4 +635,6 @@ workflows:
           requires:
             - build-linux
       - test-windows
-      - test-mac
+      - test-mac:
+          requires:
+            - build-linux
```
Comment (kripken) [inline .circleci/config.yml:640]: Might be worth a comment here, "max build depends on linux build" seems surprising...

**Review by kripken — APPROVED** [review]
Saving a third of CI time seems worth it to me!

**Pull request merged**